plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -300,18 +364,22 @@ mod tests {
     use test_case::test_case;
 
     #[test_case(
-    "aptdaemon/focal-updates,focal-updates,focal-security,focal-security,now 1.1.1+bzr982-0ubuntu32.3 all [installed]",
-    "aptdaemon", "1.1.1+bzr982-0ubuntu32.3"
+    "zsh\t5.8-6+deb11u1\tDebian Zsh Maintainers <pkg-zsh-devel@lists.alioth.debian.org>\tinstall ok installed",
+    "zsh", "5.8-6+deb11u1"
     ; "installed"
     )]
-    #[test_case(
-    "language-pack-gnome-en-base/focal-updates,focal-updates 1:20.04+20220818 all [upgradable from: 1:20.04+20220211]",
-    "language-pack-gnome-en-base", "1:20.04+20220211"
-    ; "upgradable"
-    )]
     fn get_package_name_and_version(line: &str, expected_name: &str, expected_version: &str) {
         let (name, version) = get_name_and_version(line);
         assert_eq!(name, expected_name);
         assert_eq!(version, expected_version);
     }
+
+    #[test]
+    fn both_filters_are_empty_strings() {
```

Comment:
Thanks for adding that. Now, we need test coverage for the following cases as well:

1. name: Some(...), maintainer: Some(...)
2. name: None, maintainer: Some(...)
3. name: Some(...), maintainer: None
4. name: None, maintainer: None
5. name: Some(""), maintainer: Some("")
6. name: Some(...), maintainer: Some("")
7. name: Some(""), maintainer: Some(...)

Since the filtering itself can't be tested with unit tests here, you probably will have to rely on some integration tests as you really need to execute this plugin and parse the output. Here is a test with which you can get started: https://github.com/thin-edge/thin-edge.io/tree/main/tests/RobotFramework/tests/plugin_apt